llvm/test/Assembler/invalid-c-style-comment1.ll

```diff
@@ -0,0 +1,10 @@
+; RUN: not llvm-as --disable-output %s 2>&1 | FileCheck %s -DFILE=%s
+
+@B = external global i32
+
+*/
+
+/*   /* Nested comments not supported */
+
+; CHECK: [[FILE]]:[[@LINE+1]]:1: error: redefinition of global '@B'
```

Comment:
This is not the error I would have expected for this test case. Why isn't there an error at the first `*/`?

Reply:
WDYM? Even in C: "/* hello /* this is comment */ " is a full comment. The second /* is ignored, and the comment ends at the first */. Clang does give a warnings:

```
comment.c:2:10: warning: '/*' within block comment [-Wcomment]
/* hello /* this is comment */
```

but accepts this code. Are you suggesting that we be strict and disallow /* within the /* */ comment?

Reply:
Looking at Clang Lexer, it has:

```
      // If this is a /* inside of the comment, emit a warning.  Don't do this
      // if this is a /*/, which will end the comment.  This misses cases with
      // embedded escaped newlines, but oh well.```
```

So if we add the `/*` within `/*...*/` comment check, we just need to handle this case of allowing `/*/` to terminate the comment.